CMakeLists.txt

```diff
@@ -82,38 +53,68 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
         find_package(Boost COMPONENTS regex)
         if(NOT Boost_FOUND)
             message(STATUS "GCC less then 4.9 and boost-regex NOT found - no regex used")
-            target_compile_definitions(json-schema-validator PRIVATE -DJSON_SCHEMA_NO_REGEX)
+            target_compile_definitions(nlohmann_json_schema_validator PRIVATE -DJSON_SCHEMA_NO_REGEX)
         else()
             message(STATUS "GCC less then 4.9 and boost-regex FOUND - using boost::regex")
-            target_compile_definitions(json-schema-validator PRIVATE -DJSON_SCHEMA_BOOST_REGEX)
-            target_include_directories(json-schema-validator PRIVATE ${Boost_INCLUDE_DIRS})
-            target_link_libraries(json-schema-validator PRIVATE ${Boost_LIBRARIES})
+            target_compile_definitions(nlohmann_json_schema_validator PRIVATE -DJSON_SCHEMA_BOOST_REGEX)
+            target_include_directories(nlohmann_json_schema_validator PRIVATE ${Boost_INCLUDE_DIRS})
+            target_link_libraries(nlohmann_json_schema_validator PRIVATE ${Boost_LIBRARIES})
         endif()
     endif()
 endif()
 
-if(NOT TARGET json-hpp) # if used as a subdirectory do not install json-schema.hpp
-    install(
-        FILES
-           ${CMAKE_CURRENT_SOURCE_DIR}/src/json-schema.hpp
-        DESTINATION
-            ${CMAKE_INSTALL_PREFIX}/include
-    )
-endif()
+install(TARGETS nlohmann_json_schema_validator
+        EXPORT ${PROJECT_NAME}Targets
+        LIBRARY DESTINATION lib
+        ARCHIVE DESTINATION lib
+        RUNTIME DESTINATION bin)
+
+install(FILES src/nlohmann/json-schema.hpp
+        DESTINATION include/nlohmann)
 
 if (BUILD_EXAMPLES)
-    # simple json-schema-validator-executable
+    # simple nlohmann_json_schema_validator-executable
     add_executable(json-schema-validate app/json-schema-validate.cpp)
-    target_link_libraries(json-schema-validate json-schema-validator)
+    target_link_libraries(json-schema-validate nlohmann_json_schema_validator)
 
     add_executable(readme-json-schema app/readme.cpp)
-    target_link_libraries(readme-json-schema json-schema-validator)
+    target_link_libraries(readme-json-schema nlohmann_json_schema_validator)
 endif()
 
-#add_subdirectory(ng)
-
 if (BUILD_TESTS)
     # test-zone
     enable_testing()
     add_subdirectory(test)
 endif()
+#---------------------------------------------------------------------------##
+# Set Up the Project Targets and Config Files for CMake
+#---------------------------------------------------------------------------##
+
+# Set the install path to the cmake config files
+set(INSTALL_CMAKE_DIR ${CMAKE_INSTALL_PREFIX}/lib/cmake/${PROJECT_NAME})
+
+# Create the ConfigVersion file
+include(CMakePackageConfigHelpers) # write_basic_package_version_file
+write_basic_package_version_file( ${PROJECT_NAME}ConfigVersion.cmake
+                                  VERSION ${PACKAGE_VERSION}
+                                  COMPATIBILITY SameMajorVersion)
```

Comment:
```suggestion
                                  COMPATIBILITY AnyNewerVersion)
```

Reply:
Version 2.1.0 will not be backwards compatible since the names of the libraries change as well as the install directory. Therefore, you should require the version be 2.1.0 or greater for compatibility.